C/src/xplaneConnect.c

```diff
@@ -584,23 +585,31 @@ int sendPOSI(XPCSocket sock, float values[], int size, char ac)
 	}
 
 	// Setup command
-	// 5 byte header + up to 7 values * 5 bytes each
-	unsigned char buffer[40] = "POSI";
+	unsigned char buffer[46] = "POSI";
+	buffer[4] = 0xff; //Placeholder for message length
 	buffer[5] = ac;
     int i; // iterator
-	for (i = 0; i < 7; i++)
+
+	for (i = 0; i < 7; i++) // double for lat/lon/h
 	{
-		float val = -998;
+		double val = -998;
 
 		if (i < size)
 		{
 			val = values[i];
 		}
-		*((float*)(buffer + 6 + i * 4)) = val;
+		if (i < 3) /* lat/lon/h */
+		{
+			*((double*)(buffer + 6 + i * 8)) = val;
```

Comment:
Although this follows the current code, I would actually like to see these changed to user `memcpy`. Fancy pointer casting works fine on x86, but since I originally wrote this code I've learned that this is actually undefined behavior, and it will crash on ARM processors.